identity/selector/handler_test.go

```diff
@@ -46,102 +46,43 @@ func TestUseExistingSucceeds(t *testing.T) {
 	assert.Equal(t, "pass", identityManager.LastUnlockPassphrase)
 }
 
-func TestUseExistingFailsWhenUnlockFails(t *testing.T) {
+func TestUUseOrCreateFailsWhenUnlockFails(t *testing.T) {
 	identityManager := identity.NewIdentityManagerFake([]identity.Identity{existingIdentity}, newIdentity)
 	identityManager.MarkUnlockToFail()
 	registry := &mockRegistry{}
 	cache := identity.NewIdentityCacheFake()
 
 	handler := NewHandler(identityManager, registry, cache, fakeSignerFactory)
 
-	_, err := handler.UseExisting(existingIdentity.Address, "pass")
+	_, err := handler.UseOrCreate(existingIdentity.Address, "pass")
 	assert.Error(t, err)
 
 	assert.Equal(t, existingIdentity.Address, identityManager.LastUnlockAddress)
 	assert.Equal(t, "pass", identityManager.LastUnlockPassphrase)
 }
 
-func TestUseFailsWhenIdentityNotFound(t *testing.T) {
-	identityManager := identity.NewIdentityManagerFake([]identity.Identity{existingIdentity}, newIdentity)
-	registry := &mockRegistry{}
-	cache := identity.NewIdentityCacheFake()
-
-	handler := NewHandler(identityManager, registry, cache, fakeSignerFactory)
-
-	_, err := handler.UseExisting("does-not-exist", "pass")
-	assert.NotNil(t, err)
-}
-
-func TestUseLastSucceeds(t *testing.T) {
-	identityManager := identity.NewIdentityManagerFake([]identity.Identity{existingIdentity}, newIdentity)
-	registry := &mockRegistry{}
-	cache := identity.NewIdentityCacheFake()
-
-	fakeIdentity := identity.FromAddress("abc")
-	cache.StoreIdentity(fakeIdentity)
-
-	handler := NewHandler(identityManager, registry, cache, fakeSignerFactory)
-
-	id, err := handler.UseLast("pass")
-	assert.Equal(t, fakeIdentity, id)
-	assert.Nil(t, err)
-
-	assert.Equal(t, "", registry.registeredIdentity.Address)
-
-	assert.Equal(t, "abc", identityManager.LastUnlockAddress)
-	assert.Equal(t, "pass", identityManager.LastUnlockPassphrase)
-}
-
-func TestUseLastFailsWhenUnlockFails(t *testing.T) {
-	identityManager := identity.NewIdentityManagerFake([]identity.Identity{existingIdentity}, newIdentity)
-	identityManager.MarkUnlockToFail()
-	registry := &mockRegistry{}
-	cache := identity.NewIdentityCacheFake()
-
-	fakeIdentity := identity.FromAddress("abc")
-	cache.StoreIdentity(fakeIdentity)
-
-	handler := NewHandler(identityManager, registry, cache, fakeSignerFactory)
-
-	_, err := handler.UseLast("pass")
-	assert.Error(t, err)
-
-	assert.Equal(t, "", registry.registeredIdentity.Address)
-
-	assert.Equal(t, "abc", identityManager.LastUnlockAddress)
-	assert.Equal(t, "pass", identityManager.LastUnlockPassphrase)
-}
-
-func TestUseNewSucceeds(t *testing.T) {
-	identityManager := identity.NewIdentityManagerFake([]identity.Identity{existingIdentity}, newIdentity)
+func TestUseOrCreateReturnsFirstIdentity(t *testing.T) {
```

Comment:
U can test private functions, instead of removing tests

Reply:
How? they're not private functions, they're private struct methods:/

Reply:
Is it an OK practice to make things public just for the sake of testing? @Waldz @soffokl

Reply:
Nevermind, Waldz showed me my mistake:)